### ISSUE TITLE
Let's Encrypt certificates

### ISSUE THREAD

**Issue by espadrine** [badges/shields#1460]
We currently rely on CloudFlare [Full SSL](https://www.cloudflare.com/ssl) for the TLS certificates for shields.io and img.shields.io.

It means that the individual servers use self-signed certificates, which CloudFlare connects to, and CloudFlare produces its own certificates that apply to shields.io and \*.shields.io for inbound traffic from the rest of the Internet.

Using authentic Let's Encrypt certificates instead of the self-signed ones may finally make the monitoring endpoint work. It currently fails because the websocket connection expects a valid certificate.

```bash
apt install certbot python-pip
pip install certbot certbot-dns-cloudflare
# TODO insert export env variables
certbot certonly --email XXXX -d shields.io -d img.shields.io --dns-cloudflare --test-cert
```

Note: Let's Encrypt does not yet support wildcard DNS; see <https://letsencrypt.org/2017/07/06/wildcard-certificates-coming-jan-2018.html>.

Current issue:

```bash
$ certbot plugins
An unexpected error occurred:
VersionConflict: (certbot 0.19.0 (/usr/lib/python2.7/dist-packages), Requirement.parse('certbot==0.21.0'))
Please see the logfile '/tmp/tmpSay7_c' for more details.
$ lsb_release -a | grep Release
Release:	16.04
```

---

Related: https://github.com/badges/shields/issues/608

**Comment by Daniel15** [comment]
> Note: Let's Encrypt does not yet support wildcard DNS;

They do now! I've tested wildcard DNS with Let's Encrypt and they work great :) 
![](http://ss.dan.cx/2018/04/Certificate_Viewer_%E2%80%9Cdl.vc%E2%80%9D_15-12.55.10.png)

I'm using [acme-dns](https://github.com/joohoi/acme-dns) and [certbot](https://certbot.eff.org/) 0.22.2

**Comment by paulmelnikow** [comment]
Resolved via #4929. 😌
